locals.tf

```diff
@@ -23,8 +23,8 @@ locals {
   }
 
   config_dns = merge(var.config_dns, { irsa = {
-    service_account = try(var.config_dns.irsa.service_account, "external-dns")
-    namespace       = try(var.config_dns.irsa.namespace, "external-dns")
+    external_dns = [for sa in try(var.config_dns.irsa.external-dns, []) : sa]
```

Comment:
The second param should be `["external-dns:external-dns"]` right?

Reply:
This line can just be `external_dns = try(var.config_dns.irsa.external-dns, ["external-dns:external-dns"])`